google-cloud-storage/src/test/java/com/google/cloud/storage/it/RequestAuditing.java

```diff
@@ -65,21 +65,33 @@ void clear() {
   }
 
   void assertQueryParam(String paramName, String expectedValue) {
+    assertQueryParam(paramName, ImmutableList.of(expectedValue), Function.identity());
+  }
+
+  void assertQueryParam(String paramName, ImmutableList<String> expectedValue) {
     assertQueryParam(paramName, expectedValue, Function.identity());
   }
 
-  <T> void assertQueryParam(String paramName, T expectedValue, Function<String, T> transform) {
+  <T> void assertQueryParam(String paramName, T expected, Function<String, T> transform) {
+    assertQueryParam(paramName, ImmutableList.of(expected), transform);
+  }
+
+  private <T> void assertQueryParam(
+      String paramName, ImmutableList<T> expected, Function<String, T> transform) {
     ImmutableList<HttpRequest> requests = getRequests();
 
     List<T> actual =
         requests.stream()
             .map(HttpRequest::getUrl)
-            .distinct() // todo: figure out why requests seem to be recorded twice for blob create
+            // When a multipart (http, not MPU) request is sent it will show up as multiple requests
```

Comment:
In this case it's due to the multiple parts, there will be a request for url `/path` for the object json, and a request for url `/path` for the content bytes. Here `distinct()` will collapse those to a single entry rather than two. I'm not totally sure why the internal structure of the requests is done like this, but it is (for multipart I would have expected some type of content rather than a whole new request but 🤷🏻‍♂️ )